### PR TITLE
feat: Completed list contacts UI

### LINKED ISSUE #4: Create list contacts UI

### PR DESCRIPTION
**Pull request by snehithlal**
Fixes #4 
- Completed Contact listing view
- Completed Delete contact action
- Overrides table UI

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Contacts/DeleteAlert.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+}) => {
+  const [deleting, setDeleting] = useState(false);
+
+  const handleDelete = () => {
+    setDeleting(true);
+    setContacts(prevContacts =>
+      prevContacts.filter(contact => contact.id !== selectedContactId)
+    );
+    Toastr.success("Contact deleted succesfully");
+    setSelectedContactId();
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/DeleteAlert.jsx:19]: Not sure why am empty setter is being called. This is an anti-pattern. All setters are supposed to be passed in some value.

File: app/javascript/src/components/Dashboard/Contacts/index.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import DeleteAlert from "./DeleteAlert";
+import Menubar from "./Menubar";
+import Table from "./Table";
+
+const Contacts = () => {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [showMenubar, setShowMenuBar] = useState(true);
+  const [contacts, setContacts] = useState(CONTACTS);
+  const [showDeleteAlert, setShowDeleteAlert] = useState(false);
+  const [selectedContactId, setSelectedContactId] = useState();
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/index.jsx:16]: Big Anti-pattern. No state should be declared without initial values.

**Review by yedhink — COMMENTED** [review]
@snehithlal Few comments

File: app/javascript/src/components/Dashboard/Contacts/index.jsx (hunk 2 above)
Comment (snehithlal) [reply app/javascript/src/components/Dashboard/Contacts/index.jsx:16]: @yedhink  Fixed with #14

**Pull request merged**